libappstream-glib/as-app-validate.c

```diff
@@ -471,10 +472,15 @@ ai_app_validate_image_check (AsImage *im, AsAppValidateHelper *helper)
 
 	/* send sync */
 	status_code = soup_session_send_message (helper->session, msg);
-	if (status_code != SOUP_STATUS_OK) {
+	if (SOUP_STATUS_IS_TRANSPORT_ERROR(status_code)) {
 		ai_app_validate_add (helper,
 				     AS_PROBLEM_KIND_URL_NOT_FOUND,
-				     "<screenshot> url not found [%s]", url);
+				     "<screenshot> failed to connect: %s [%s] ", soup_status_get_phrase(status_code), url);
+		return FALSE;
+	} else if (status_code != SOUP_STATUS_OK) {
+		ai_app_validate_add (helper,
+				     AS_PROBLEM_KIND_URL_NOT_FOUND,
+				     "<screenshot> HTTP %d: %s [%s]", status_code, soup_status_get_phrase(status_code), url);
```

Comment:
maybe something like `<screenshot> failed to download, HTTP error was %s [%d]`

Also, both added lines need wrapping to 80ish chars wide. Thanks.